2-standard-library/activityMonitor.js

```diff
@@ -0,0 +1,56 @@
+const childProcess = require('child_process');
+const fs = require('fs');
+
+const logFilePath = 'activityMonitor.log';
+
+let logData = '';
+
+const getCurrentTime = () => {
+    return Math.floor(Date.now() / 1000);
+}
+
+const getProcessCommand = () => {
+    if (process.platform === 'win32') {
+        return 'powershell "Get-Process | Sort-Object CPU -Descending | Select-Object -Property Name, CPU, WorkingSet -First 1 | ForEach-Object { $_.Name + \' \' + $_.CPU + \' \' + $_.WorkingSet }"';
+    } else {
+        return 'ps -A -o %cpu,%mem,comm | sort -nr | head -n 1';
+    }
+}
+
+const logSystemActivity = (data) => {
+    fs.appendFile(logFilePath, data, (err) => {
+        if (err) throw err;
+    });
+}
+
+const truncateText = (text, maxlength) => {
+    if (text.length > maxlength) {
+        return text.slice(0, maxlength - 3) + '...';
+    }
+    return text;
+}
+
+const monitorSystem = () => {
```

Comment:
You can simplify this function by the following:
1. use the async/awit pattern here
2. use of promisify() from the util module of node.js to create a promise-based version of exec().
3. await the outcome of exec(command).
4. replace the callback logic with a try-catch block to handle errors. I have a commented it as well

Reply:
Done